src/quantforge/backtest_strategy.py

```diff
@@ -0,0 +1,597 @@
+                             row = next_day_slice.loc[ticker_str]
+                             # Create OHLCData object - Ensure NaN handling if needed
+                             ohlc = OHLCData(
+                                 date=next_ts.date(), # Use date part? Check OHLCData definition
+                                 open=row[OPEN],
+                                 high=row[HIGH],
+                                 low=row[LOW],
+                                 close=row[CLOSE],
+                                 # volume=row.get(VOLUME, None) # Check if OHLCData includes volume
+                             )
+                             next_day_data[item_obj] = ohlc
+                except KeyError:
+                    # No data available for next_ts for some/all tickers
+                     print(f"Warning: No data found for next timestamp {next_ts.date()}")
+                except Exception as e:
+                    print(f"Error preparing next_day_data for {next_ts.date()}: {e}")
+
+
+            # 3. Update Portfolio's current prices (needed for valuation)
+            # Store these prices within the Backtest object
+            current_prices_dict: Dict[TradeableItem, float] = {}
+            try:
+                latest_data_this_ts = self.raw_data.xs(current_ts, level=TIMESTAMP)
+                for ticker_str, item_obj in self.tradeable_items_map.items():
+                    if ticker_str in latest_data_this_ts.index:
+                        close_price = latest_data_this_ts.loc[ticker_str, CLOSE]
+                        if pd.notna(close_price):
+                            current_prices_dict[item_obj] = close_price
+                # Update the stored prices in the Backtest class
+                self.current_market_prices = current_prices_dict 
+            except KeyError:
+                 print(f"Warning: No data found for current timestamp {current_ts.date()}")
+            except Exception as e:
+                print(f"Error updating portfolio prices for {current_ts.date()}: {e}")
+
+
+            # 4. Execute strategy logic
+            try:
+                 self.strategy.execute(input_data, next_day_data)
+            except Exception as e:
+                 print(f"!!! Error during strategy execution on {current_ts.date()}: {e}")
+                 import traceback
+                 traceback.print_exc() # Print stack trace for debugging
+
+
+            # 5. Record portfolio value at the end of the day
+            # Use the updated current market prices for end-of-day valuation
+            current_total_value = self.portfolio.portfolio_value(self.current_market_prices)
+            self.equity_history.append((current_ts, current_total_value))
+
+
+        print("Backtest simulation finished.")
+        # Calculate and return performance metrics
+        metrics = self.get_performance_metrics()
+        return metrics
+
+    # --- Metrics Calculation (Needs Adaptation) ---
```

Comment:
remove all metrics calculation for now into its own file. We will have a whole set of metrics calculation.